hw/ip/otbn/dv/otbnsim/sim/sim.py

```diff
@@ -175,11 +175,29 @@ def _step_idle(self, verbose: bool) -> StepRes:
             self.state.ext_regs.write('INSN_CNT', 0, True)
 
         if self.state.init_sec_wipe_is_running():
-            # Wait for the URND seed. Once that appears, switch to WIPING_GOOD
-            # unless the FSM state is already LOCKED, in which case we change
-            # it to WIPING_BAD.
+            # Wait for the URND seed. If there is some genuine state to wipe
+            # (because self.state.has_state_to_wipe is true), then we switch to
+            # WIPING_GOOD unless the FSM state is already LOCKED, in which case
+            # we change it to WIPING_BAD.
+            #
+            # As a special case, there might be an RMA request and this might
+            # be the start of time so we don't actually have any state to wipe.
+            # In that case, we jump straight to the LOCKED state.
```

Comment:
I think the code does the right thing but the reasoning is a bit different:
- After a reset we ideally always reseed URND and then to do a secure wipe. There can be state because for FI hardening reasons, we've removed the resets from the wide register file (see https://opentitan.org/book/doc/security/implementation_guidelines/reset_vs_non-reset_flops/index.html#appendix-otbn---a-case-for-non-reset-flops to get some insight on the motivation).
- If we get an RMA request right after reset, we don't perform the URND reseed. Because the RMA request typically comes before the entropy complex is set up. So the URND reseed might never finish and thus the wipe never happen and thus RMA never entered.
- Not doing the full secure wipe (including URND) is acceptable from a security perspective in the case of RMA entry only because:
  - Triggering the RMA entry is itself hardened.
  - After RMA entry, the chip resets and while booting into RMA, another full secure wipe including URND reseed happens.

Reply:
Thanks. I've tweaked the text to be a slightly looser version of what you're saying :-)